code/object/object.cpp

```diff
@@ -1289,9 +1290,82 @@ void obj_move_all_post(object *objp, float frametime)
 					}
 				}
 				if (cast_light) {
-					weapon_info* wi = &Weapon_info[Weapons[objp->instance].weapon_info_index];
+					weapon* wp = &Weapons[objp->instance];
+					weapon_info* wi = &Weapon_info[wp->weapon_info_index];
 					auto lp = lighting_profiles::current();
 					hdr_color light_color;
+
+					float intensity_mult = 1.f;
+					float radius_mult = 1.f;
+					float r_mult = 1.f;
+					float g_mult = 1.f;
+					float b_mult = 1.f;
+
+					for (uint c = 0; c < wi->lifetime_curves.size(); c++) {
+						WeaponLifetimeCurve* mod_curve = &wi->lifetime_curves[c];
+						if (mod_curve->curve_idx < 0) {
+							Warning(LOCATION, "Curve does not exist!");
+							continue;
+						}
+
+						Curve curve = Curves[mod_curve->curve_idx];
+						float input = 1.0f;
+						float output = 1.0f;
+						switch (mod_curve->input) {
+							case WeaponLifetimeCurveInput::LIFETIME:
+								input = f2fl(Missiontime - wp->creation_time) / wi->lifetime;
+								break;
+							case WeaponLifetimeCurveInput::AGE:
+								input = f2fl(Missiontime - wp->creation_time);
+								break;
+							case WeaponLifetimeCurveInput::BASE_VELOCITY:
+								input = wp->weapon_max_vel;
+								break;
+							case WeaponLifetimeCurveInput::HEALTH:
```

Comment:
Very pedantic, but I'd prefer "HP" or "Hitpoints". The engine, with rare exception, doesn't really refer to it as "Health". Goes for all the inputs and such as well.